src/components/menu/menu-interim-element.js

```diff
@@ -0,0 +1,307 @@
+          positionMenu(scope, element, opts);
+          $$rAF(function() {
+            element.addClass('md-active');
+            opts.alreadyOpen = true;
+            element[0].style[$mdConstant.CSS.TRANSFORM] = '';
+          });
+        });
+      });
+
+      return $mdUtil.transitionEndPromise(element, {timeout: 350});
+
+
+      // Activate interaction on the menu popup, allowing it to be closed by
+      // clicking on the backdrop, with escape, clicking options, etc.
+      function activateInteraction() {
+        element.addClass('md-clickable');
+        opts.backdrop && opts.backdrop.on('click', function(e) {
+          e.preventDefault();
+          e.stopPropagation();
+          opts.mdMenuCtrl.close(true);
+        });
+
+        opts.menuContentEl.on('keydown', function(ev) {
+          scope.$apply(function() {
+            switch (ev.keyCode) {
+              case $mdConstant.KEY_CODE.ESCAPE: opts.mdMenuCtrl.close(); break;
+              case $mdConstant.KEY_CODE.UP_ARROW: focusPrevMenuItem(ev, opts.menuContentEl, opts); break;
+              case $mdConstant.KEY_CODE.DOWN_ARROW: focusNextMenuItem(ev, opts.menuContentEl, opts); break;
+            }
+          });
+        });
+
+        opts.menuContentEl.on('click', function(e) {
+          var target = e.target;
+          do {
+            if (target && target.hasAttribute('ng-click')) {
+              if (!target.hasAttribute('disabled')) {
+                close();
+              }
+              break;
+            }
+          } while ((target = target.parentNode) && target != opts.menuContentEl)
+
+          function close() {
+            scope.$apply(function() {
+              opts.mdMenuCtrl.close();
+            });
+          }
+        });
+
+        var focusTarget = opts.menuContentEl[0].querySelector('[md-menu-focus-target]');
+        if (!focusTarget) focusTarget = opts.menuContentEl[0].firstElementChild.firstElementChild;
+        focusTarget.focus();
+      }
+    }
+
+    function focusPrevMenuItem(e, menuEl, opts) {
+      var currentItem = $mdUtil.getClosest(e.target, 'MD-MENU-ITEM');
+
+      var items = nodesToArray(menuEl[0].children);
+      var currentIndex = items.indexOf(currentItem);
+
+      for (var i = currentIndex - 1; i >= 0; --i) {
+        var focusTarget = items[i].firstElementChild || items[i];
+        var didFocus = attemptFocus(focusTarget);
+        if (didFocus) {
+          break;
+        }
+      }
+    }
+
+    function focusNextMenuItem(e, menuEl, opts) {
+      var currentItem = $mdUtil.getClosest(e.target, 'MD-MENU-ITEM');
+
+      var items = nodesToArray(menuEl[0].children);
+
+      var currentIndex = items.indexOf(currentItem);
+
+      for (var i = currentIndex + 1; i < items.length; ++i) {
+        var focusTarget = items[i].firstElementChild || items[i];
+        var didFocus = attemptFocus(focusTarget);
+        if (didFocus) {
+          break;
+        }
+      }
+    }
+
+    function attemptFocus(el) {
```

Comment:
This function in particular could use a description of what it does (attempt?) and what its return value means.
(though I'm a fan of all functions having at least one short sentence explaning what they do)